### PR TITLE
PR for Code Reviewing (Not To be Checked In)

### PR DESCRIPTION
**Pull request by jessicl-ms**
## Checklist

- [ ] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [ ] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorBackendPoolsSettingsObject.cs [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+    /// </summary>
+    [Cmdlet("New", ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "FrontDoorBackendPoolsSettingsObject"), OutputType(typeof(PSBackendPoolsSettings))]
+    public class NewAzureRmBackendPoolsSettingsObject : AzureFrontDoorCmdletBase
+    {
+        /// <summary>
+        /// Gets or sets the BackendPoolSettings name.
+        /// </summary>
+        [Parameter(Mandatory = true, HelpMessage = "BackendPoolsSettings name.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorBackendPoolsSettingsObject.cs]: >Name [](start = 22, length = 4)

Is the name defined in swagger? #Resolved

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoor.cs [hunk 2]
```diff
@@ -106,6 +114,23 @@ public override void ExecuteCmdlet()
...
+            {
+                case FieldsWithCertificateNameCheckParameterSet:
+                {
+                    backendPoolsSettings = new Management.FrontDoor.Models.BackendPoolsSettings(
+                            DisableCertificateNameCheck ? PSEnforceCertificateNameCheck.Disabled.ToString() : PSEnforceCertificateNameCheck.Enabled.ToString());
+                    break;
+                }
+                default:
+                {
+                    backendPoolsSettings = BackendPoolsSettings?.ToSdkBackendPoolsSettings();
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoor.cs]: >backendPoolsSettings [](start = 20, length = 20)

What happens if backendpoolssettings are not passed? Will RP handle null backendpoolsSettings? #Resolved

Comment (pichandwork) [reply src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoor.cs]: Add a test case for this.

---
In reply to: [330339054](https://github.com/jessicl-ms/azure-powershell/pull/1#discussion_r330339054) [](ancestors = 330339054)

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs [hunk 3]
```diff
@@ -57,47 +52,53 @@ public class NewAzureRmFrontDoorFrontendEndpointObject : AzureFrontDoorCmdletBas
...
+        /// <summary>
+        /// Defines the TLS extension protocol that is used for secure delivery. Part of CustomHttpsConfiguration.
         /// </summary>
         [Parameter(Mandatory = false, HelpMessage = "The TLS extension protocol that is used for secure delivery")]
         [PSArgumentCompleter("ServerNameIndication", "IPBased")]
         public string ProtocolType { get; set; }
 
         /// <summary>
-        /// Defines the TLS extension protocol that is used for secure delivery
+        /// Defines the TLS extension protocol that is used for secure delivery. Part of CustomHttpsConfiguration.
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: >Defines the TLS extension protocol that is used for secure delivery [](start = 12, length = 67)

incorrect comment #Resolved

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs [hunk 4]
```diff
@@ -109,9 +104,15 @@ public class SetAzureRmFrontDoor : AzureFrontDoorCmdletBase
         /// <summary>
         /// Whether to enforce certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.
         /// </summary>
-        [Parameter(Mandatory = false, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
+        [Parameter(Mandatory = false, HelpMessage = "Settings for all backendPools")]
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: >Settings for all backendPools [](start = 53, length = 29)

incoorect help #Resolved

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs [hunk 5]
```diff
@@ -109,9 +104,15 @@ public class SetAzureRmFrontDoor : AzureFrontDoorCmdletBase
...
         public SwitchParameter DisableCertificateNameCheck { get; set; }
 
+        /// <summary>
+        /// Settings for all backendPools
+        /// </summary>
+        [Parameter(Mandatory = false, HelpMessage = "Whether to disable certificate name check on HTTPS requests to all backend pools. No effect on non-HTTPS requests.")]
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: >HelpMessage [](start = 38, length = 11)

incorrect help #Resolved

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs [hunk 6]
```diff
@@ -186,9 +187,20 @@ public override void ExecuteCmdlet()
                 updateParameters.EnabledState = EnabledState;
             }
 
-            if (this.IsParameterBound(c => c.DisableCertificateNameCheck))
+            // Should not be able to specify both DisableCertificateNameCheck and pass in full BackendPoolsSettings object
+            // But the existence of these fields are optional and not bound to a specific parameter set
+            if (this.IsParameterBound(c => c.BackendPoolsSettings) && this.IsParameterBound(c => c.DisableCertificateNameCheck))
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: >if (this.IsParameterBound(c => c.BackendPoolsSettings) && this.IsParameterBound(c => c.DisableCertificateNameCheck)) [](start = 12, length = 116)

Why not use parameterset similar to New-AzFrontDoor? #Resolved

**Comment by pichandwork** [comment]
> ### Example 1: Create a Front Door based on given parameters.

Update the example

---
Refers to: src/FrontDoor/FrontDoor/help/New-AzFrontDoor.md:38 in 6646154. [](commit_id = 6646154192f96e050801e9925a4713bd598d07b3, deletion_comment = False)

File: src/FrontDoor/FrontDoor/help/New-AzFrontDoorBackendPoolsSettingsObject.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+external help file: Microsoft.Azure.PowerShell.Cmdlets.FrontDoor.dll-Help.xml
+Module Name: Az.FrontDoor
+online version:
+schema: 2.0.0
+---
+
+# New-AzFrontDoorBackendPoolsSettingsObject
+
+## SYNOPSIS
+{{ Fill in the Synopsis }}
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/New-AzFrontDoorBackendPoolsSettingsObject.md]: Synopsis, Description etc need to be filled. Example need to be added #Resolved

**Comment by pichandwork** [comment]
> ### Example 1

Update the example and also output

---
Refers to: src/FrontDoor/FrontDoor/help/New-AzFrontDoorFrontendEndpointObject.md:28 in 6646154. [](commit_id = 6646154192f96e050801e9925a4713bd598d07b3, deletion_comment = False)

**Comment by pichandwork** [comment]
> ### Example 1

Update the example and output

---
Refers to: src/FrontDoor/FrontDoor/help/New-AzFrontDoorHealthProbeSettingObject.md:26 in 6646154. [](commit_id = 6646154192f96e050801e9925a4713bd598d07b3, deletion_comment = False)

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md [hunk 8]
```diff
@@ -26,19 +26,37 @@ Set-AzFrontDoor -ResourceGroupName <String> -Name <String> [-RoutingRule <PSRout
 Set-AzFrontDoor -InputObject <PSFrontDoor> [-RoutingRule <PSRoutingRule[]>] [-BackendPool <PSBackendPool[]>]
  [-FrontendEndpoint <PSFrontendEndpoint[]>] [-LoadBalancingSetting <PSLoadBalancingSetting[]>]
  [-HealthProbeSetting <PSHealthProbeSetting[]>] [-Tag <Hashtable>] [-EnabledState <PSEnabledState>]
- [-DisableCertificateNameCheck] [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
- [<CommonParameters>]
+ [-DisableCertificateNameCheck] [-BackendPoolsSettings <PSBackendPoolsSettings>]
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md]: >[-DisableCertificateNameCheck] [-BackendPoolsSettings <PSBackendPoolsSettings>] [](start = 1, length = 79)

This seems incorrect. Cant have both DisableCertificateNameCheck and BackendPoolsSettings in the same parameter set. #Closed

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md [hunk 9]
```diff
@@ -26,19 +26,37 @@ Set-AzFrontDoor -ResourceGroupName <String> -Name <String> [-RoutingRule <PSRout
...
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
 ```
 
 ### ByResourceIdParameterSet
 ```
 Set-AzFrontDoor -ResourceId <String> [-RoutingRule <PSRoutingRule[]>] [-BackendPool <PSBackendPool[]>]
+ [-FrontendEndpoint <PSFrontendEndpoint[]>] [-LoadBalancingSetting <PSLoadBalancingSetting[]>]
+ [-HealthProbeSetting <PSHealthProbeSetting[]>] [-Tag <Hashtable>] [-EnabledState <PSEnabledState>]
+ [-DisableCertificateNameCheck] [-BackendPoolsSettings <PSBackendPoolsSettings>]
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md]: >[-DisableCertificateNameCheck] [-BackendPoolsSettings <PSBackendPoolsSettings>] [](start = 1, length = 79)

This seems incorrect. Cant have both DisableCertificateNameCheck and BackendPoolsSettings in the same parameter set. #Resolved

File: src/FrontDoor/FrontDoor/Helpers/ModelExtensions.cs [hunk 10]
```diff
@@ -284,18 +302,29 @@ public static SdkLoadBalancingSetting ToSdkLoadBalancingSetting(this PSLoadBalan
...
+            if ((psFrontendEndpoint.CertificateSource != null) ||
+                !String.IsNullOrEmpty(psFrontendEndpoint.MinimumTlsVersion) ||
+                !String.IsNullOrEmpty(psFrontendEndpoint.Vault) ||
+                !String.IsNullOrEmpty(psFrontendEndpoint.SecretName) ||
+                !String.IsNullOrEmpty(psFrontendEndpoint.SecretVersion) ||
+                !String.IsNullOrEmpty(psFrontendEndpoint.CertificateType))
+            {
+                customHttpsConfiguration = new SdkHttpsConfig(psFrontendEndpoint.CertificateSource,
+                                   psFrontendEndpoint.MinimumTlsVersion,
+                                   new SdkValut(psFrontendEndpoint.Vault),
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Helpers/ModelExtensions.cs]: >SdkValut [](start = 39, length = 8)

Fix the Typo #Resolved

File: src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1 [hunk 11]
```diff
@@ -18,48 +18,76 @@ Full Front Door CRUD cycle
 #>
 function Test-FrontDoorCrud
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1:19]: >Test-FrontDoorCrud [](start = 9, length = 18)

Add a test case Test-FrontDoorCrudDefaults passing only mandatory parameters and see if the defaults are populated correctly. #Resolved

**Review by pichandwork — CHANGES_REQUESTED** [review]
:clock1:

File: src/FrontDoor/FrontDoor/Cmdlets/EnableAzureRmFrontDoorCustomDomainHttps.cs [hunk 12]
```diff
@@ -12,25 +12,21 @@
...
-using System.Linq;
 using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.FrontDoor;
 using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System.Management.Automation;
 
 namespace Microsoft.Azure.Commands.FrontDoor.Cmdlets
 {
     /// <summary>
     /// Defines the Enable-AzCustomDomainHttps cmdlet.
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/EnableAzureRmFrontDoorCustomDomainHttps.cs:28]: >Enable-AzCustomDomainHttps [](start = 20, length = 26)

Doesnt this cmdlet take min tls version? #Resolved

Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/EnableAzureRmFrontDoorCustomDomainHttps.cs:28]: Yes. I forgot to push the changes.

File: src/FrontDoor/FrontDoor/Helpers/ModelExtensions.cs (hunk 10 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Helpers/ModelExtensions.cs]: Fixed.

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs (hunk 3 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: Removed "Part of CustomHttpsConfiguration" for all parameters in CustomHttpsConfiguration.

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs (hunk 4 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: Fixed.

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs (hunk 5 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: Fixed.

File: src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs (hunk 6 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/SetAzureRmFrontDoor.cs]: When I tried to do that, I hit issues with PowerShell not being able to resolve the ParameterSet, but I think I wasn't implementing that correctly. Will try again.

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs [hunk 13]
```diff
@@ -107,15 +108,16 @@ public override void ExecuteCmdlet()
...
                 SessionAffinityTtlSeconds = !this.IsParameterBound(c => c.SessionAffinityTtlInSeconds) ? 0 : SessionAffinityTtlInSeconds,
                 WebApplicationFirewallPolicyLink = WebApplicationFirewallPolicyLink,
                 CertificateSource = CertificateSource,
                 CertificateType = CertificateType,
                 Vault = Vault,
                 SecretName = SecretName,
                 SecretVersion = SecretVersion,
-                ProtocolType = ProtocolType
+                ProtocolType = this.IsParameterBound(c => c.ProtocolType) ? ProtocolType : "ServerNameIndication",
+                MinimumTlsVersion = !this.IsParameterBound(c => c.SessionAffinityTtlInSeconds) ? "1.2" : MinimumTlsVersion
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: >SessionAffinityTtlInSeconds [](start = 66, length = 27)

mintlsversion

**Comment by pichandwork** [comment]
> ### Example 3: Enable HTTPS for a custom domain with PSFrontendEndpoint object using Front Door managed certificate.

Update example output to include mintlsversion output and add an example for min tls version #Resolved

---
Refers to: src/FrontDoor/FrontDoor/help/Enable-AzFrontDoorCustomDomainHttps.md:116 in bc842b2. [](commit_id = bc842b2e4f181108383a335d7f872bf627d8d6f7, deletion_comment = False)

**Comment by pichandwork** [comment]
> EnforceCertificateNameCheck : Enabled

The output for the command would change.Correct?

---
Refers to: src/FrontDoor/FrontDoor/help/New-AzFrontDoor.md:45 in bc842b2. [](commit_id = bc842b2e4f181108383a335d7f872bf627d8d6f7, deletion_comment = False)

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md [hunk 14]
```diff
@@ -14,31 +14,83 @@ Update a Front Door load balancer
 
 ### ByFieldsParameterSet (Default)
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:15]: >ByFieldsParameterSet [](start = 4, length = 20)

This parameterset have neitherdisablecertnamecheck nor backendpoolsettings?

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md [hunk 15]
```diff
@@ -14,31 +14,83 @@ Update a Front Door load balancer
...
+### ByFieldsWithBackendPoolsSettingsParameterSet
+```
+Set-AzFrontDoor -ResourceGroupName <String> -Name <String> [-RoutingRule <PSRoutingRule[]>]
+ [-BackendPool <PSBackendPool[]>] [-FrontendEndpoint <PSFrontendEndpoint[]>]
+ [-LoadBalancingSetting <PSLoadBalancingSetting[]>] [-HealthProbeSetting <PSHealthProbeSetting[]>]
+ [-Tag <Hashtable>] [-EnabledState <PSEnabledState>] -BackendPoolsSettings <PSBackendPoolsSettings>
+ [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm] [<CommonParameters>]
+```
+
 ### ByObjectParameterSet
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:42]: >ByObjectParameterSet [](start = 4, length = 20)

This parameterset have neitherdisablecertnamecheck nor backendpoolsettings?

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md [hunk 16]
```diff
@@ -14,31 +14,83 @@ Update a Front Door load balancer
...
+### ByObjectWithBackendPoolsSettingsParameterSet
+```
+Set-AzFrontDoor -InputObject <PSFrontDoor> [-RoutingRule <PSRoutingRule[]>] [-BackendPool <PSBackendPool[]>]
+ [-FrontendEndpoint <PSFrontendEndpoint[]>] [-LoadBalancingSetting <PSLoadBalancingSetting[]>]
+ [-HealthProbeSetting <PSHealthProbeSetting[]>] [-Tag <Hashtable>] [-EnabledState <PSEnabledState>]
+ -BackendPoolsSettings <PSBackendPoolsSettings> [-DefaultProfile <IAzureContextContainer>] [-WhatIf] [-Confirm]
+ [<CommonParameters>]
+```
+
 ### ByResourceIdParameterSet
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:68]: >ByResourceIdParameterSet [](start = 4, length = 24)

This parameterset have neitherdisablecertnamecheck nor backendpoolsettings?

**Comment by pichandwork** [comment]
> ### Example 1: update an existing Front Door with FrontDoorName and ResourceGroupName.

update output of examples.

---
Refers to: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:99 in bc842b2. [](commit_id = bc842b2e4f181108383a335d7f872bf627d8d6f7, deletion_comment = False)

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs (hunk 13 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: The Swagger refers to it as MinimumTlsVersion. ?

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md (hunk 16 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:68]: See previous comment.

File: src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md (hunk 14 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor/help/Set-AzFrontDoor.md:15]: Fixed Parameter Set options. Please check and see if it makes sense. The difficulty I was having before was keeping -DisableCertificateNameCheck and BackendPoolsSettings optional, but XOR when one of them was present. I took each of the existing ParameterSets (ResourceId, Object, Fields) and subdivided them into 3 other parameter sets (neither fields, only DisableCertificateNameCheck, only BackendPoolSettings.

File: src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1 (hunk 11 above)
Comment (jessicl-ms) [reply src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1:19]: Added. Let me know if you think it needs more explicit validation.

File: src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs (hunk 13 above)
Comment (pichandwork) [reply src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: I am refreing to "!this.IsParameterBound(c => c.SessionAffinityTtlInSeconds)". It shouldnt be SessionAffinityTtlInSeconds

---
In reply to: [331267175](https://github.com/jessicl-ms/azure-powershell/pull/1#discussion_r331267175) [](ancestors = 331267175)

Comment (pichandwork) [reply src/FrontDoor/FrontDoor/Cmdlets/NewAzureRmFrontDoorFrontendEndpointObject.cs]: I am refreing to "!this.IsParameterBound(c => c.SessionAffinityTtlInSeconds)". It shouldnt be SessionAffinityTtlInSeconds

---
In reply to: [330856087](https://github.com/jessicl-ms/azure-powershell/pull/1#discussion_r330856087) [](ancestors = 330856087)

File: src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1 [hunk 17]
```diff
@@ -18,48 +18,126 @@ Full Front Door CRUD cycle
...
+    Assert-AreEqual $retrievedFrontDoor.RoutingRules[0].RouteConfiguration.GetType().Name "PSForwardingConfiguration"
+    Assert-Tags $tags $retrievedFrontDoor.Tags
+
+	# Verify Default HealthProbeSettings
+	Assert-AreEqual $retrievedFrontDoor.HealthProbeSettings[0].Name $healthProbeSetting1.Name
+	Assert-AreEqual $retrievedFrontDoor.HealthProbeSettings[0].HealthProbeMethod "Head"
+	Assert-AreEqual $retrievedFrontDoor.HealthProbeSettings[0].EnabledState "Enabled"
+	Assert-AreEqual $retrievedFrontDoor.HealthProbeSettings[0].IntervalInSeconds 30
+
+	# Verify Default BackendPoolsSettings
```
Comment (pichandwork) [inline src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1]: >Verify [](start = 3, length = 6)

Also verify default min tls version.

**Comment by pichandwork** [comment]
>     New-AzFrontDoor -Name $Name -ResourceGroupName $resourceGroupName -RoutingRule $routingrule1 -BackendPool $backendpool1 -FrontendEndpoint $frontendEndpoint1 -LoadBalancingSetting $loadBalancingSetting1 -HealthProbeSetting $healthProbeSetting1 -Tag $tags

I see backendpoolsettings object is not passed. Verify if default backendpoolsettings are set.

---
Refers to: src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1:225 in 47c49f3. [](commit_id = 47c49f3b48606c1e915ec32ef7fac2e679f28485, deletion_comment = False)

**Review by jessicl-ms — COMMENTED** [review]
> > ```
> > New-AzFrontDoor -Name $Name -ResourceGroupName $resourceGroupName -RoutingRule $routingrule1 -BackendPool $backendpool1 -FrontendEndpoint $frontendEndpoint1 -LoadBalancingSetting $loadBalancingSetting1 -HealthProbeSetting $healthProbeSetting1 -Tag $tags
> > ```
> 
> I see backendpoolsettings object is not passed. Verify if default backendpoolsettings are set.
> 
> Refers to: src/FrontDoor/FrontDoor.Test/ScenarioTests/FrontDoorTests.ps1:225 in [47c49f3](https://github.com/jessicl-ms/azure-powershell/commit/47c49f3b48606c1e915ec32ef7fac2e679f28485). [](commit_id = [47c49f3](https://github.com/jessicl-ms/azure-powershell/commit/47c49f3b48606c1e915ec32ef7fac2e679f28485), deletion_comment = False)

Updated Test-FrontDoorCrudWithPiping to verify values

**Pull request closed without merging**